### PR TITLE
Feat/Listen to and process `RemoveNode` notification

### LINKED ISSUE #977: IR: Listen and process `RemoveNode` notifications of `Subnet` contract
Related to 
* nspcc-dev/neofs-contract#122
* nspcc-dev/neofs-node#974

## Summary

`RemoveNode` methods of `Subnet` contract throws notifications about nodes excluded from the subnet. Alphabetic nodes should respond to this by removing the subnet attribute from the node information as a candidate for a network map. If a node remains outside of any subnet, then it loses its candidacy.

## Task

- [ ] Define format of `RemoveNode` notification event. Implement notification parser.

- [ ] Make Netmap processor to listen for `RemoveNode` notifications from sidechain.

- [ ] If removed node is a netmap candidate then do same subnet-related steps as in nspcc-dev/neofs-node#974 but call `AddPeer` if node has been left the subnet (`UpdateState(StateOffline)` if the subnet was the only one).

### PR DESCRIPTION
**Pull request by carpawell**
Closes #977.

### CODE REVIEW

File: pkg/innerring/processors/netmap/handlers.go [hunk 1]
```diff
@@ -99,3 +100,22 @@ func (np *Processor) handleCleanupTick(ev event.Event) {
 			zap.Int("capacity", np.pool.Cap()))
 	}
 }
+
+func (np *Processor) handleRemoveNode(ev event.Event) {
+	removeNode := ev.(subnet.RemoveNode)
+
+	np.log.Info("notification",
+		zap.String("type", "remove node from subnet"),
+		zap.String("subnetID", removeNode.SubnetworkID().String()),
```
Comment (cthulhu-rider) [inline pkg/innerring/processors/netmap/handlers.go:109]:
```suggestion
		zap.Stringer("subnetID", removeNode.SubnetworkID()),
```

File: pkg/innerring/processors/netmap/process_peers.go [hunk 2]
```diff
@@ -142,3 +145,70 @@ func (np *Processor) processUpdatePeer(ev netmapEvent.UpdatePeer) {
...
+
+	candidates, err := np.netmapClient.GetCandidates()
+	if err != nil {
+		np.log.Warn("could not get network map candidates",
+			zap.Error(err),
+		)
+		return
+	}
+
+	removeFromSubnetwork := ev.SubnetworkID()
```
Comment (cthulhu-rider) [inline pkg/innerring/processors/netmap/process_peers.go:163]: Var name seems like action name. Better to name it `subnetToRemoveFrom` or smth like that.

File: pkg/innerring/processors/netmap/process_peers.go [hunk 3]
```diff
@@ -142,3 +145,70 @@ func (np *Processor) processUpdatePeer(ev netmapEvent.UpdatePeer) {
...
+
+	for _, node := range candidates.Nodes {
+		if !bytes.Equal(node.NodeInfo.PublicKey(), ev.Node().Bytes()) {
+			continue
```
Comment (cthulhu-rider) [inline pkg/innerring/processors/netmap/process_peers.go:167]: I don't see `break`. May be we'll seek the node in for-loop and process it oustside?

File: pkg/morph/event/subnet/remove_node.go [hunk 4]
```diff
@@ -0,0 +1,86 @@
...
+}
+
+// MorphEvent implements Neo:Morph Event interface.
+func (RemoveNode) MorphEvent() {}
+
+// SubnetworkID returns a marshalled subnetID structure, defined in API.
+func (rn RemoveNode) SubnetworkID() *subnetid.ID { return rn.subnetID }
+
+// Node is public key of the nodeKey that is being deleted.
+func (rn RemoveNode) Node() *keys.PublicKey { return rn.nodeKey }
```
Comment (cthulhu-rider) [inline pkg/morph/event/subnet/remove_node.go:33]: We work only with `RemoveNode{}.Node().Bytes()`, so there is no need to interpret key bytes.

File: pkg/morph/event/subnet/remove_node.go [hunk 5]
```diff
@@ -0,0 +1,86 @@
...
+	}
+
+	rawSubID, err = client.BytesFromStackItem(params[0])
+	if err != nil {
+		return nil, fmt.Errorf("could not get raw subnetID: %w", err)
+	}
+
+	ev.subnetID = new(subnetid.ID)
+
+	err = ev.subnetID.Unmarshal(rawSubID)
```
Comment (cthulhu-rider) [inline pkg/morph/event/subnet/remove_node.go:68]: Why here? This should be done on IR side. Anyway you don't check if ID refers to zero subnet.

File: pkg/innerring/processors/netmap/process_peers.go [hunk 6]
```diff
@@ -142,3 +145,70 @@ func (np *Processor) processUpdatePeer(ev netmapEvent.UpdatePeer) {
...
+		}
+
+		hasOtherSubnets := false
+
+		err = node.IterateSubnets(func(subNetID subnetid.ID) error {
+			if subNetID.Equals(removeFromSubnetwork) {
+				return netmap.ErrRemoveSubnet
+			}
+
+			hasOtherSubnets = true
```
Comment (cthulhu-rider) [inline pkg/innerring/processors/netmap/process_peers.go:177]: `node.IterateSubnets` returns error in this case. We can use `if-else` below with `err != nil` condition.

File: pkg/innerring/processors/netmap/handlers.go (hunk 1 above)
Comment (carpawell) [reply pkg/innerring/processors/netmap/handlers.go]: changed

File: pkg/innerring/processors/netmap/process_peers.go (hunk 2 above)
Comment (carpawell) [reply pkg/innerring/processors/netmap/process_peers.go]: renamed

File: pkg/innerring/processors/netmap/process_peers.go (hunk 3 above)
Comment (carpawell) [reply pkg/innerring/processors/netmap/process_peers.go:176]: added break at the end of `for` body, think that is enough

File: pkg/morph/event/subnet/remove_node.go (hunk 4 above)
Comment (carpawell) [reply pkg/morph/event/subnet/remove_node.go]: changed

File: pkg/morph/event/subnet/remove_node.go (hunk 5 above)
Comment (carpawell) [reply pkg/morph/event/subnet/remove_node.go]: deleted marshaling in the parser

File: pkg/innerring/processors/netmap/process_peers.go (hunk 6 above)
Comment (carpawell) [reply pkg/innerring/processors/netmap/process_peers.go]: changed

**Review by cthulhu-rider — APPROVED** [review]

**Pull request merged**